internal/log/log.go

```diff
@@ -59,96 +56,111 @@ type CtxKeyAccount struct{}
 
-var (
-	_ loggers.Logger = (*zap.Logger)(nil)
-	_ loggers.Logger = (*console.Logger)(nil)
-)
+const envVarLogFormat = "MONACO_LOG_FORMAT"
+const envVarLogTime = "MONACO_LOG_TIME"
 
 func Fatal(msg string, a ...interface{}) {
-	std.Fatal(msg, a...)
+	slog.Error(fmt.Sprintf(msg, a...))
+	os.Exit(1)
 }
 
 func Error(msg string, a ...interface{}) {
-	std.Error(msg, a...)
+	slog.Error(fmt.Sprintf(msg, a...))
 }
 
 func Warn(msg string, a ...interface{}) {
-	std.Warn(msg, a...)
+	slog.Warn(fmt.Sprintf(msg, a...))
 }
 
 func Info(msg string, a ...interface{}) {
-	std.Info(msg, a...)
+	slog.Info(fmt.Sprintf(msg, a...))
 }
 
 func Debug(msg string, a ...interface{}) {
-	std.Debug(msg, a...)
+	slog.Debug(fmt.Sprintf(msg, a...))
 }
 
-func Level() loggers.LogLevel {
-	return std.Level()
+// PrepareLogging sets up the default slog.Logger using the specified options.
+func PrepareLogging(ctx context.Context, fs afero.Fs, verbose bool, loggerSpy io.Writer, fileLogging bool, enableMemstatLogging bool) {
+	logger := slog.New(prepareHandler(ctx, fs, verbose, loggerSpy, fileLogging, enableMemstatLogging))
+	slog.SetDefault(logger)
 }
 
-// WithFields adds additional [field.Field] for structured logs
-// It accepts vararg fields and should not be called more than once per log call
-func WithFields(fields ...field.Field) loggers.Logger {
-	return std.WithFields(fields...)
-}
+func prepareHandler(ctx context.Context, fs afero.Fs, verbose bool, loggerSpy io.Writer, fileLogging bool, enableMemstatLogging bool) slog.Handler {
+	handlerOptions := getHandlerOptions(getLevelFromVerbose(verbose))
+	handlers := []slog.Handler{getHandler(os.Stderr, handlerOptions)}
 
-// WithCtxFields creates a logger instance with preset structured logging [field.Field] based on the Context
-// Coordinate (via [CtxKeyCoord]) and environment (via [CtxKeyEnv] [CtxValEnv]) information is added to logs from the Context
-func WithCtxFields(ctx context.Context) loggers.Logger {
-	loggr := std
-	f := make([]field.Field, 0, 2)
-	if c, ok := ctx.Value(CtxKeyCoord{}).(coordinate.Coordinate); ok {
-		f = append(f, field.Coordinate(c))
-	}
-	if e, ok := ctx.Value(CtxKeyEnv{}).(CtxValEnv); ok {
-		f = append(f, field.Environment(e.Name, e.Group))
+	if loggerSpy != nil {
+		handlers = append(handlers, getHandler(loggerSpy, handlerOptions))
 	}
 
-	if a := ctx.Value(CtxKeyAccount{}); a != nil {
-		f = append(f, field.F("account", a))
+	if fileLogging && fs != nil {
+		logFile, errorFile, err := PrepareLogFiles(ctx, fs, enableMemstatLogging)
+		if err != nil {
+			Warn("Error preparing log files: %s", err.Error())
+		}
+
+		if logFile != nil {
+			handlers = append(handlers, getHandler(logFile, handlerOptions))
+		}
+
+		if errorFile != nil {
+			handlers = append(handlers, getHandler(errorFile, getHandlerOptions(slog.LevelError)))
+		}
 	}
 
-	if c, ok := ctx.Value(CtxGraphComponentId{}).(CtxValGraphComponentId); ok {
-		f = append(f, field.F("gid", c))
+	if len(handlers) == 1 {
+		return handlers[0]
```

Comment:
I guess there isn't an easy way to have just one handler? For multiple write operations, `io.MultiWriter` could be used. Not sure how to handle the errorFile handler though

Reply:
For the moment, this is the easiest way as we want different things to go to the different sinks in different formats. For example, in a future PR, we'd like color logs on the console, but plain text going to the files, and as you say, only errors and fatals going to the error file, but all logs going to the log file. Something to revisit this in the future, though, and optimization is certainly possible.  

One last thought: in a future PR, our handler might also add attributes that it pulls from the context. The TeeHandler could do this once, and then, with this current architecture, pass this on a handler that supports color, or, encodes it to JSON, saving us from having to reimplement all this functionality.